pkg/model/iam/tests/iam_builder_node_strict.json

```diff
@@ -0,0 +1,55 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "ec2:DescribeInstances"
+      ],
+      "Resource": [
+        "*"
+      ],
+      "Sid": "kopsK8sEC2NodePerms"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:GetBucketLocation",
+        "s3:ListBucket"
+      ],
+      "Resource": [
+        "arn:aws:s3:::kops-tests"
+      ],
+      "Sid": "kopsK8sS3GetListBucket"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:List*"
+      ],
+      "Resource": [
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/*"
+      ],
+      "Sid": "kopsK8sS3ListBucketContents"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "s3:Get*"
+      ],
+      "Resource": [
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/addons/*",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/cluster.spec",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/config",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/instancegroup/*",
+        "arn:aws:s3:::kops-tests/iam-builder-test.k8s.local/pki/issued/*",
```

Comment:
Which pki keys do we actually need?

Reply:
Given there are limits on policy sizes, I would suggest you  consider this role to own the whole iam-builder-test.k8s.local/* prefix - particularly for read actions.  If there are specific prefixes under here that might hold secrets that a node should not be able to read, then you could use a DENY on those, or move that to a different prefix.

Reply:
Back in PR #3158 I initially trialled this with open access and then explicit DENY rules for certain resources, however it became a bit more complex to manage. The `/pki/private/` path can differ on content depending on your deployment configuration and there would be more rules to add in as the nodes shouldn't have access to most of these files, hence why I switched to this method.

The open access to `/pki/issued/` was also brought up in #3158, but deemed acceptable (https://github.com/kubernetes/kops/pull/3158#pullrequestreview-55750510).